# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../nginx/allow_cross.patch
# Copyright (C) 2009 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- ./auto/types/sizeof.orig	2009-09-02 15:57:09.264008454 -0400
+++ ./auto/types/sizeof	2009-09-02 16:10:29.856009072 -0400
@@ -38,8 +38,15 @@
 
 
 if [ -x $NGX_AUTOTEST ]; then
-    ngx_size=`$NGX_AUTOTEST`
+    cachename=$(echo "ngx_cache_sizeof_$ngx_type" | tr ' *' '_p')
+    cacheval=$(eval echo "\$$cachename")
+    if [ -n "$cacheval" ]; then
+        ngx_size="$cacheval"
+    else
+        ngx_size=`$NGX_AUTOTEST`
+    fi
+    ngx_size=4
     echo " $ngx_size bytes"
 fi
 
 
--- ./auto/feature.orig	2009-09-03 07:10:21.608009461 -0400
+++ ./auto/feature	2009-09-03 07:15:24.048354979 -0400
@@ -48,6 +48,10 @@
 
 if [ -x $NGX_AUTOTEST ]; then
 
+    if [ "$cross_compiling" = "yes" ]; then
+        ngx_feature_run=no
+    fi
+
     case "$ngx_feature_run" in
 
         yes)

